src/ProjectorMonteCarloProblem.jl

```diff
@@ -0,0 +1,261 @@
+"""
+    FCIQMC()
+Algorithm for the full configuration interaction quantum Monte Carlo (FCIQMC) method.
+To be passed as the `algorithm` keyword argument to [`ProjectorMonteCarloProblem`](@ref).
+"""
+struct FCIQMC end
+
+"""
+    SimulationPlan(; starting_step = 1, last_step = 100, walltime = Inf)
+Defines the duration of the simulation. The simulation ends when the `last_step` is reached
+or the `walltime` is exceeded.
+
+See [`ProjectorMonteCarloProblem`](@ref), [`PMCSimulation`](@ref).
+"""
+Base.@kwdef struct SimulationPlan
+    starting_step::Int = 0
+    last_step::Int = 100
+    walltime::Float64 = Inf
+end
+
+"""
+    ProjectorMonteCarloProblem(hamiltonian::AbstractHamiltonian; kwargs...)
+Defines a problem to be solved by projector quantum Monte Carlo (QMC) methods, such as the
+the [`FCIQMC`](@ref) algorithm.
+
+# Common keyword arguments and defaults:
+- `time_step = 0.01`: Initial time step size.
+- `last_step = 100`: Controls the number of steps.
+- `targetwalkers = 1_000`: Target for the 1-norm of the coefficient vector.
+- `start_at = starting_address(hamiltonian)`: Define the initial state vector. This can be a
+    single address, a collection of addresses, a single starting vector, or a collection of
+    starting vectors. If multiple starting vectors are passed, their number must match the
+    number of replicas. If (a collection of) [`AbstractDVec`](@ref)(s) is passed, the
+    keyword arguments `style`, `initiator`, and `threading` are ignored.
+- `style = IsDynamicSemistochastic()`: The [`StochasticStyle`](@ref) of the simulation.
+- `initiator = NonInitiator()`: Whether to use initiators. See [`InitiatorRule`](@ref).
+- `threading`: Default is to use multithreading and
+  [MPI](https://juliaparallel.org/MPI.jl/latest/) if multiple threads are available. Set to
+  `true` to force [`PDVec`](@ref) for the starting vector, `false` for serial computation;
+  may be overridden by `start_at`.
```

Comment:
This makes it sound like MPI is used if threading is available. Maybe this is clearer?

```suggestion
- `threading`: Default is to use multithreading and/or
  [MPI](https://juliaparallel.org/MPI.jl/latest/) if available. Set to
  `true` to force [`PDVec`](@ref) for the starting vector, `false` for serial computation;
  may be overridden by `start_at`.
```